applications/nrf_desktop/src/modules/hids.c

```diff
@@ -430,28 +430,41 @@ static void send_hid_report(const struct hid_report_event *event)
 
 	switch (report_id) {
 	case REPORT_ID_BOOT_MOUSE:
-		err = bt_hids_boot_mouse_inp_rep_send(&hids_obj, cur_conn,
-						      &buffer[0], buffer[1],
-						      buffer[2],
-						      report_sent_cb[report_id]
-						      );
+		if (!protocol_boot) {
+			err = -EBADF;
+		} else {
+			err = bt_hids_boot_mouse_inp_rep_send(&hids_obj, cur_conn,
+							      &buffer[0], buffer[1],
+							      buffer[2],
+							      report_sent_cb[report_id]);
+		}
 		break;
 	case REPORT_ID_BOOT_KEYBOARD:
-		err = bt_hids_boot_kb_inp_rep_send(&hids_obj, cur_conn,
-						   buffer, size,
-						   report_sent_cb[report_id]);
+		if (!protocol_boot) {
+			err = -EBADF;
+		} else {
+			err = bt_hids_boot_kb_inp_rep_send(&hids_obj, cur_conn,
+							   buffer, size,
+							   report_sent_cb[report_id]);
+		}
 		break;
 	default:
-		err = bt_hids_inp_rep_send(&hids_obj, cur_conn,
-					   report_index[report_id],
-					   buffer, size,
-					   report_sent_cb[report_id]);
+		if (protocol_boot) {
+			err = -EBADF;
+		} else {
+			err = bt_hids_inp_rep_send(&hids_obj, cur_conn,
+						   report_index[report_id],
+						   buffer, size,
+						   report_sent_cb[report_id]);
+		}
 		break;
 	}
 
 	if (err) {
 		if (err == -ENOTCONN) {
```

Comment:
Similar race condition as for boot protocol could occur if host disables notifications for given HID report ID.
In that case we would get `-EACESS` as an error code if we try to send report. Maybe we should handle it too and print warning instead of error log to prevent failures in CI in that case?

Reply:
Let's leave it for now. This is not the same situation. Change of the protocol type was not handled while this one is. If our CI will cover tests that will require log severity downgrade we will do the change. At the moment it's not important really.